test/test_multithread_stress.c

```diff
@@ -272,9 +279,39 @@ int main(int argc, char **argv)
 		__atomic_load(&failed_thread, &abort, __ATOMIC_RELAXED);
 	}
 
-	for (int i = 0; i < thread_num; i++) {
-		pthread_join(thread_info[i].tid, NULL);
-	}
+	int try_again=0;
+	gettimeofday(&start_time, NULL);
+	do {
```

Comment:
@tuliom  The code will only get here after all threads have returned, so this timeout logic won't help much here. This loop should be combined with the previous one (`while (abort == 0)`) to have the desired effect.

Reply:
@mscastanho I disagree with you both when you say this logic won't help much and when you say this will be reached only after all threads have returned.
Line 284 is reached in 2 scenarios:

1. When at least 1 thread fails. In this case, all other threads are likely running.
2. When all threads have been **started** `test_iterations` times. In this case, we have between 0 and all threads still running.

In both cases, it does make sense to kill the remaining threads if they do not complete in 5 minutes, but I'm more interested on case 1 because I've seen it triggering issues where threads do not exit after hours of execution.
Case 2 is just expected behavior and, as such, should be completed in less than 5 minutes too.

I do agree with you this is not helping against an uncaught issue in lines 262-272, but I'm afraid it's hard to set such a value when the number of threads, test interval and the number of iterations vary. IMHO, this would be better served with a watchdog monitoring each test independently, i.e. each libnxz test process.
This will require further work and it is not my goal at the moment.

Reply:
Makes sense, thanks for the clarification.